src/CBT.NuGet.AggregatePackage.UnitTests/AggregateTest.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            IDictionary<string, string> pkgInstalls = new Dictionary<string, string>();
+            foreach (var sourceNupkg in Directory.EnumerateFiles(repo, filePattern))
+            {
+                var destPath = Path.Combine(destinationRoot, Path.GetFileNameWithoutExtension(sourceNupkg));
+                ZipFile.ExtractToDirectory(sourceNupkg, destPath);
+                var nuspec = Directory.EnumerateFiles(destPath, "*.nuspec", SearchOption.TopDirectoryOnly).SingleOrDefault();
+                var packageID = Path.GetFileNameWithoutExtension(nuspec);
+
+                pkgInstalls.Add(packageID, destPath);
+            }
+            return pkgInstalls;
+        }
+
+        private void CreateTestProject(string testProject)
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(testProject));
+            var testProjectElement = ProjectBuilder.Create()
+                    .AddProperty("CBTEnablePackageRestore=true")
+                    .AddProperty("Platform=AnyCPU", "Configuration=Debug", "OutputPath=bin")
+                    .AddProperty($"CBTNuGetTasksAssemblyPath={Assembly.GetExecutingAssembly().Location}")
+                    .AddProperty("CBTAggregatePackage=NuGetPath_Microsoft_Build=src1|scr2")
+                    .AddProperty($"NuGetPackagesPath={_destPackagesPath}")
+                    .AddProperty($"IntermediateOutputPath={_enlistmentRoot}\\obj\\$(Configuration)\\$(Platform)\\AggregatePackage.proj")
+                    .AddImport($"{_installedNupkgs["CBT.NuGet.AggregatePackage"]}\\cbt\\module\\After.CBT.Nuget.props")
+                    .AddImport($"{_installedNupkgs["CBT.NuGet.AggregatePackage"]}\\cbt\\module\\After.Microsoft.Common.targets")
+                    .AddTarget("Build")
+                    .Save(testProject);
+        }
+
+        private void CreateAggregateImportProject(string project)
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(project));
+            ProjectBuilder.Create()
+                    .AddProperty("CBTAggregatePackage=NuGetPath_Microsoft_Build=src1|src2")
+                    .Save(project);
+        }
+
+        private void SetupTestEnlistment(string enlistmentRoot, IDictionary<string, string> installedNupkgs)
+        {
+            _testProject = Path.Combine(_enlistmentRoot, "src", "AggregatePackage.proj");
+            CreateTestProject(_testProject);
+
+            string aggregateImport = Path.Combine(_enlistmentRoot, "src", "CBT.AggregatePackages.props");
+            CreateAggregateImportProject(aggregateImport);
+
+            _aggregatePropsInNupkg = Path.Combine(installedNupkgs["CBT.NuGet.AggregatePackage"], "CBT", "Module", "CBT.Nuget.AggregatePackage.props");
+            File.WriteAllText(_aggregatePropsInNupkg, File.ReadAllText(_aggregatePropsInNupkg).Replace("CBT.NuGet.Tasks.AggregatePackages", "CBT.NuGet.AggregatePackage.UnitTests.FakeAggregateTrue"));
```

Comment:
Put some comments in this method to explain what its doing so other people will understand.